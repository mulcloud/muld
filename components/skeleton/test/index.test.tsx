import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import { render, cleanup, fireEvent } from '@testing-library/react';
import 'regenerator-runtime';
import Skeleton from '..';

const LoadingDemo = () => {
    const [showLoading, setShowLoading] = React.useState(true);
    return (
        <React.Fragment>
            <button
                onClick={() => {
                    setShowLoading(!showLoading);
                }}
            >
                显示
            </button>
            <Skeleton title avatar row="3" loading={showLoading}>
                <div className="actual">实际内容</div>
            </Skeleton>
        </React.Fragment>
    );
};

describe('Skeleton', () => {
    afterEach(cleanup);
    it('title prop', () => {
        const { asFragment } = render(<Skeleton title={true} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('round prop', () => {
        const { asFragment } = render(<Skeleton round={true} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('animate prop', () => {
        const { asFragment } = render(<Skeleton animate={false} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('loading prop', () => {
        const { asFragment } = render(<Skeleton loading={true} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('avatar prop', () => {
        const { asFragment } = render(<Skeleton avatar={true} avatarSize="30" />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('avatarSize prop', () => {
        const { asFragment } = render(<Skeleton avatar avatarSize="30" />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('avatarShape prop', () => {
        const { asFragment } = render(<Skeleton avatar avatarShape="square" />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('row & rowWidth & titleWidth prop string', () => {
        const { asFragment } = render(<Skeleton row="10" rowWidth="20" titleWidth="30" />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('row & rowWidth & titleWidth prop number', () => {
        const { asFragment } = render(<Skeleton rowWidth={30} row={5} titleWidth={30} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('rowWidth prop Array', () => {
        const { asFragment } = render(<Skeleton rowWidth={['10', 20, 30]} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('loading-change prop', () => {
        const { getByText, container } = render(<LoadingDemo />);
        const button = getByText('显示');
        fireEvent.click(button);
        const actual = container.querySelector('.actual')!;
        expect(actual).toBeInTheDocument();
    });
});
