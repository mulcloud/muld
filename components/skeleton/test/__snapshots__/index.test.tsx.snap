// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton animate prop 1`] = `
<DocumentFragment>
  .c0.mul-skeleton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 16px;
}

.c0.mul-skeleton .mul-skeleton__avatar {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__avatar--round {
  border-radius: 999px;
}

.c0.mul-skeleton .mul-skeleton__content {
  width: 100%;
}

.c0.mul-skeleton .mul-skeleton__avatar + .mul-skeleton__content {
  padding-top: 8px;
}

.c0.mul-skeleton .mul-skeleton__row,
.c0.mul-skeleton .mul-skeleton__title {
  height: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__title {
  margin: 0;
}

.c0.mul-skeleton .mul-skeleton__row:not(:first-child) {
  margin-top: 12px;
}

.c0.mul-skeleton .mul-skeleton__title + .mul-skeleton__row {
  margin-top: 20px;
}

.c0.mul-skeleton--animate {
  -webkit-animation: mul-skeleton-blink 1.2s ease-in-out infinite;
  animation: mul-skeleton-blink 1.2s ease-in-out infinite;
}

.c0.mul-skeleton--round .mul-skeleton__row,
.c0.mul-skeleton--round .mul-skeleton__title {
  border-radius: 999px;
}

<div
    class="c0 mul-skeleton"
  >
    <div
      class="mul-skeleton__content"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton avatar prop 1`] = `
<DocumentFragment>
  .c0.mul-skeleton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 16px;
}

.c0.mul-skeleton .mul-skeleton__avatar {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__avatar--round {
  border-radius: 999px;
}

.c0.mul-skeleton .mul-skeleton__content {
  width: 100%;
}

.c0.mul-skeleton .mul-skeleton__avatar + .mul-skeleton__content {
  padding-top: 8px;
}

.c0.mul-skeleton .mul-skeleton__row,
.c0.mul-skeleton .mul-skeleton__title {
  height: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__title {
  margin: 0;
}

.c0.mul-skeleton .mul-skeleton__row:not(:first-child) {
  margin-top: 12px;
}

.c0.mul-skeleton .mul-skeleton__title + .mul-skeleton__row {
  margin-top: 20px;
}

.c0.mul-skeleton--animate {
  -webkit-animation: mul-skeleton-blink 1.2s ease-in-out infinite;
  animation: mul-skeleton-blink 1.2s ease-in-out infinite;
}

.c0.mul-skeleton--round .mul-skeleton__row,
.c0.mul-skeleton--round .mul-skeleton__title {
  border-radius: 999px;
}

<div
    class="c0 mul-skeleton mul-skeleton--animate"
  >
    <div
      class="mul-skeleton__avatar mul-skeleton__avatar--round"
      style="width: 30px; height: 30px;"
    />
    <div
      class="mul-skeleton__content"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton avatarShape prop 1`] = `
<DocumentFragment>
  .c0.mul-skeleton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 16px;
}

.c0.mul-skeleton .mul-skeleton__avatar {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__avatar--round {
  border-radius: 999px;
}

.c0.mul-skeleton .mul-skeleton__content {
  width: 100%;
}

.c0.mul-skeleton .mul-skeleton__avatar + .mul-skeleton__content {
  padding-top: 8px;
}

.c0.mul-skeleton .mul-skeleton__row,
.c0.mul-skeleton .mul-skeleton__title {
  height: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__title {
  margin: 0;
}

.c0.mul-skeleton .mul-skeleton__row:not(:first-child) {
  margin-top: 12px;
}

.c0.mul-skeleton .mul-skeleton__title + .mul-skeleton__row {
  margin-top: 20px;
}

.c0.mul-skeleton--animate {
  -webkit-animation: mul-skeleton-blink 1.2s ease-in-out infinite;
  animation: mul-skeleton-blink 1.2s ease-in-out infinite;
}

.c0.mul-skeleton--round .mul-skeleton__row,
.c0.mul-skeleton--round .mul-skeleton__title {
  border-radius: 999px;
}

<div
    class="c0 mul-skeleton mul-skeleton--animate"
  >
    <div
      class="mul-skeleton__avatar mul-skeleton__avatar--square"
      style="width: 32px; height: 32px;"
    />
    <div
      class="mul-skeleton__content"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton avatarSize prop 1`] = `
<DocumentFragment>
  .c0.mul-skeleton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 16px;
}

.c0.mul-skeleton .mul-skeleton__avatar {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__avatar--round {
  border-radius: 999px;
}

.c0.mul-skeleton .mul-skeleton__content {
  width: 100%;
}

.c0.mul-skeleton .mul-skeleton__avatar + .mul-skeleton__content {
  padding-top: 8px;
}

.c0.mul-skeleton .mul-skeleton__row,
.c0.mul-skeleton .mul-skeleton__title {
  height: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__title {
  margin: 0;
}

.c0.mul-skeleton .mul-skeleton__row:not(:first-child) {
  margin-top: 12px;
}

.c0.mul-skeleton .mul-skeleton__title + .mul-skeleton__row {
  margin-top: 20px;
}

.c0.mul-skeleton--animate {
  -webkit-animation: mul-skeleton-blink 1.2s ease-in-out infinite;
  animation: mul-skeleton-blink 1.2s ease-in-out infinite;
}

.c0.mul-skeleton--round .mul-skeleton__row,
.c0.mul-skeleton--round .mul-skeleton__title {
  border-radius: 999px;
}

<div
    class="c0 mul-skeleton mul-skeleton--animate"
  >
    <div
      class="mul-skeleton__avatar mul-skeleton__avatar--round"
      style="width: 30px; height: 30px;"
    />
    <div
      class="mul-skeleton__content"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton loading prop 1`] = `
<DocumentFragment>
  .c0.mul-skeleton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 16px;
}

.c0.mul-skeleton .mul-skeleton__avatar {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__avatar--round {
  border-radius: 999px;
}

.c0.mul-skeleton .mul-skeleton__content {
  width: 100%;
}

.c0.mul-skeleton .mul-skeleton__avatar + .mul-skeleton__content {
  padding-top: 8px;
}

.c0.mul-skeleton .mul-skeleton__row,
.c0.mul-skeleton .mul-skeleton__title {
  height: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__title {
  margin: 0;
}

.c0.mul-skeleton .mul-skeleton__row:not(:first-child) {
  margin-top: 12px;
}

.c0.mul-skeleton .mul-skeleton__title + .mul-skeleton__row {
  margin-top: 20px;
}

.c0.mul-skeleton--animate {
  -webkit-animation: mul-skeleton-blink 1.2s ease-in-out infinite;
  animation: mul-skeleton-blink 1.2s ease-in-out infinite;
}

.c0.mul-skeleton--round .mul-skeleton__row,
.c0.mul-skeleton--round .mul-skeleton__title {
  border-radius: 999px;
}

<div
    class="c0 mul-skeleton mul-skeleton--animate"
  >
    <div
      class="mul-skeleton__content"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton round prop 1`] = `
<DocumentFragment>
  .c0.mul-skeleton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 16px;
}

.c0.mul-skeleton .mul-skeleton__avatar {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__avatar--round {
  border-radius: 999px;
}

.c0.mul-skeleton .mul-skeleton__content {
  width: 100%;
}

.c0.mul-skeleton .mul-skeleton__avatar + .mul-skeleton__content {
  padding-top: 8px;
}

.c0.mul-skeleton .mul-skeleton__row,
.c0.mul-skeleton .mul-skeleton__title {
  height: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__title {
  margin: 0;
}

.c0.mul-skeleton .mul-skeleton__row:not(:first-child) {
  margin-top: 12px;
}

.c0.mul-skeleton .mul-skeleton__title + .mul-skeleton__row {
  margin-top: 20px;
}

.c0.mul-skeleton--animate {
  -webkit-animation: mul-skeleton-blink 1.2s ease-in-out infinite;
  animation: mul-skeleton-blink 1.2s ease-in-out infinite;
}

.c0.mul-skeleton--round .mul-skeleton__row,
.c0.mul-skeleton--round .mul-skeleton__title {
  border-radius: 999px;
}

<div
    class="c0 mul-skeleton mul-skeleton--animate mul-skeleton--round"
  >
    <div
      class="mul-skeleton__content"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton row & rowWidth & titleWidth prop number 1`] = `
<DocumentFragment>
  .c0.mul-skeleton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 16px;
}

.c0.mul-skeleton .mul-skeleton__avatar {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__avatar--round {
  border-radius: 999px;
}

.c0.mul-skeleton .mul-skeleton__content {
  width: 100%;
}

.c0.mul-skeleton .mul-skeleton__avatar + .mul-skeleton__content {
  padding-top: 8px;
}

.c0.mul-skeleton .mul-skeleton__row,
.c0.mul-skeleton .mul-skeleton__title {
  height: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__title {
  margin: 0;
}

.c0.mul-skeleton .mul-skeleton__row:not(:first-child) {
  margin-top: 12px;
}

.c0.mul-skeleton .mul-skeleton__title + .mul-skeleton__row {
  margin-top: 20px;
}

.c0.mul-skeleton--animate {
  -webkit-animation: mul-skeleton-blink 1.2s ease-in-out infinite;
  animation: mul-skeleton-blink 1.2s ease-in-out infinite;
}

.c0.mul-skeleton--round .mul-skeleton__row,
.c0.mul-skeleton--round .mul-skeleton__title {
  border-radius: 999px;
}

<div
    class="c0 mul-skeleton mul-skeleton--animate"
  >
    <div
      class="mul-skeleton__content"
    >
      <div
        class="mul-skeleton__row"
        style="width: 30px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 30px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 30px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 30px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 30px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Skeleton row & rowWidth & titleWidth prop string 1`] = `
<DocumentFragment>
  .c0.mul-skeleton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 16px;
}

.c0.mul-skeleton .mul-skeleton__avatar {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__avatar--round {
  border-radius: 999px;
}

.c0.mul-skeleton .mul-skeleton__content {
  width: 100%;
}

.c0.mul-skeleton .mul-skeleton__avatar + .mul-skeleton__content {
  padding-top: 8px;
}

.c0.mul-skeleton .mul-skeleton__row,
.c0.mul-skeleton .mul-skeleton__title {
  height: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__title {
  margin: 0;
}

.c0.mul-skeleton .mul-skeleton__row:not(:first-child) {
  margin-top: 12px;
}

.c0.mul-skeleton .mul-skeleton__title + .mul-skeleton__row {
  margin-top: 20px;
}

.c0.mul-skeleton--animate {
  -webkit-animation: mul-skeleton-blink 1.2s ease-in-out infinite;
  animation: mul-skeleton-blink 1.2s ease-in-out infinite;
}

.c0.mul-skeleton--round .mul-skeleton__row,
.c0.mul-skeleton--round .mul-skeleton__title {
  border-radius: 999px;
}

<div
    class="c0 mul-skeleton mul-skeleton--animate"
  >
    <div
      class="mul-skeleton__content"
    >
      <div
        class="mul-skeleton__row"
        style="width: 20px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 20px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 20px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 20px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 20px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 20px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 20px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 20px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 20px;"
      />
      <div
        class="mul-skeleton__row"
        style="width: 20px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Skeleton rowWidth prop Array 1`] = `
<DocumentFragment>
  .c0.mul-skeleton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 16px;
}

.c0.mul-skeleton .mul-skeleton__avatar {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__avatar--round {
  border-radius: 999px;
}

.c0.mul-skeleton .mul-skeleton__content {
  width: 100%;
}

.c0.mul-skeleton .mul-skeleton__avatar + .mul-skeleton__content {
  padding-top: 8px;
}

.c0.mul-skeleton .mul-skeleton__row,
.c0.mul-skeleton .mul-skeleton__title {
  height: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__title {
  margin: 0;
}

.c0.mul-skeleton .mul-skeleton__row:not(:first-child) {
  margin-top: 12px;
}

.c0.mul-skeleton .mul-skeleton__title + .mul-skeleton__row {
  margin-top: 20px;
}

.c0.mul-skeleton--animate {
  -webkit-animation: mul-skeleton-blink 1.2s ease-in-out infinite;
  animation: mul-skeleton-blink 1.2s ease-in-out infinite;
}

.c0.mul-skeleton--round .mul-skeleton__row,
.c0.mul-skeleton--round .mul-skeleton__title {
  border-radius: 999px;
}

<div
    class="c0 mul-skeleton mul-skeleton--animate"
  >
    <div
      class="mul-skeleton__content"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton title prop 1`] = `
<DocumentFragment>
  .c0.mul-skeleton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 16px;
}

.c0.mul-skeleton .mul-skeleton__avatar {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__avatar--round {
  border-radius: 999px;
}

.c0.mul-skeleton .mul-skeleton__content {
  width: 100%;
}

.c0.mul-skeleton .mul-skeleton__avatar + .mul-skeleton__content {
  padding-top: 8px;
}

.c0.mul-skeleton .mul-skeleton__row,
.c0.mul-skeleton .mul-skeleton__title {
  height: 16px;
  background-color: #f2f3f5;
}

.c0.mul-skeleton .mul-skeleton__title {
  margin: 0;
}

.c0.mul-skeleton .mul-skeleton__row:not(:first-child) {
  margin-top: 12px;
}

.c0.mul-skeleton .mul-skeleton__title + .mul-skeleton__row {
  margin-top: 20px;
}

.c0.mul-skeleton--animate {
  -webkit-animation: mul-skeleton-blink 1.2s ease-in-out infinite;
  animation: mul-skeleton-blink 1.2s ease-in-out infinite;
}

.c0.mul-skeleton--round .mul-skeleton__row,
.c0.mul-skeleton--round .mul-skeleton__title {
  border-radius: 999px;
}

<div
    class="c0 mul-skeleton mul-skeleton--animate"
  >
    <div
      class="mul-skeleton__content"
    >
      <h3
        class="mul-skeleton__title"
        style="width: 40%;"
      />
    </div>
  </div>
</DocumentFragment>
`;
